quinn-proto/src/connection/streams.rs

```diff
@@ -70,6 +70,13 @@ pub struct Streams {
     send_window: u64,
     /// Configured upper bound for how much unacked data the peer can send us per stream
     stream_receive_window: u64,
+    /// Whether the corresponding `max_remote` has increased
+    max_streams_dirty: [bool; 2],
```

Comment:
This could also use the same mechanism as `max_data` and thereby be `sent_max_data`

Reply:
Punting on this until after #952 due to the error case as discussed in https://github.com/quinn-rs/quinn/pull/961#discussion_r550723332.

Reply:
Do you want to do this as part of this PR?

Reply:
Because these frames can become queued due to receiving packets, this will be a bit fiddly to pursue. My inclination is not to block the behavioral fix on that.